sigridci/sigridci/azure_pull_request_report.py

```diff
@@ -31,14 +32,21 @@ def generate(self, analysisId, feedback, options):
         if not self.isSupported(options) or not os.path.exists(feedbackFile):
             return
 
-        UploadLog.log("Sending feedback to Azure DevOps API")
-
-        request = urllib.request.Request(self.buildURL(), self.buildRequestBody(feedbackFile))
-        request.add_header("Authorization", f"Bearer {os.environ['SYSTEM_ACCESSTOKEN']}")
-        request.add_header("Content-Type", "application/json")
         try:
-            urllib.request.urlopen(request)
-            UploadLog.log("Published feedback to Azure DevOps")
+            UploadLog.log("Sending feedback to Azure DevOps API")
+
+            # We want to update the existing comment, to avoid spamming people with new
+            # comments every time they make a commit. We have no way to persist this,
+            # so we need to check the existing comments.
+            existingId = self.findExistingSigridCommentThreadId()
+            status = Objective.determineStatus(feedback, options)
+
+            if existingId == None:
+                self.callAzure("POST", self.buildRequestBody(feedbackFile, status), None)
+                UploadLog.log("Published feedback to Azure DevOps")
+            else:
+                self.callAzure("PATCH", self.buildRequestBody(feedbackFile, status), existingId)
+                UploadLog.log("Updated existing feedback in Azure DevOps")
         except urllib.error.HTTPError as e:
             UploadLog.log(f"Warning: Azure DevOps API error: {e.code} / {e.fp.read()}")
```

Comment:
How about logging even more:
```suggestion
            UploadLog.log(f"Warning: Azure DevOps API error calling {e.url}: {e.code} ({e.reason}) / {e.fp.read()}")
```

By including `{e.url}`, we know which endpoint is involved (and we learn whether we constructed the URL correctly). `{e.reason}` might give information not in the body (not sure).